Add worker_name helper to dist_utils.

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#34162 Add worker_name helper to dist_utils.**

This avoids the "worker{}".format(..) in our unit tests to something
cleaner.

Differential Revision: [D20233533](https://our.internmc.facebook.com/intern/diff/D20233533/)

pritamdamania committed:
Add worker_name helper to dist_utils.

This avoids the "worker{}".format(..) in our unit tests to something
cleaner.

Differential Revision: [D20233533](https://our.internmc.facebook.com/intern/diff/D20233533/)

[ghstack-poisoned]

pritamdamania committed:
Update on "Add worker_name helper to dist_utils."


This avoids the "worker{}".format(..) in our unit tests to something
cleaner.

Differential Revision: [D20233533](https://our.internmc.facebook.com/intern/diff/D20233533/)

[ghstack-poisoned]

pritamdamania committed:
Update on "Add worker_name helper to dist_utils."


This avoids the "worker{}".format(..) in our unit tests to something
cleaner.

Differential Revision: [D20233533](https://our.internmc.facebook.com/intern/diff/D20233533/)

[ghstack-poisoned]

wanchaol commented on torch/testing/_internal/distributed/rpc/jit/rpc_test.py:
```diff
@@ -568,25 +568,59 @@ def test_torchscript_functions_not_supported(self):
 
         # rpc_sync still accepts script class and run it in
         # the same code path as python call.
+<<<<<<< dest:   1586723a7ec3 - flato: [carbon] encrypt password in Unity in c...
```
merge conflicts here.

pritamdamania committed:
Update on "Add worker_name helper to dist_utils."


This avoids the "worker{}".format(..) in our unit tests to something
cleaner.

Differential Revision: [D20233533](https://our.internmc.facebook.com/intern/diff/D20233533/)

[ghstack-poisoned]
